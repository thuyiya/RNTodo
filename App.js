import React, { useState } from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, SafeAreaView } from 'react-native';
import { Auth, Home } from './src';



export default function App() {
  const [access, setAccess] = useState(false)

  const screenRender = () => {
    return access ? <Home /> :  <Auth setAccess={(value) => setAccess(value)}/>;
  }

  return (
    <SafeAreaView style={styles.container}>
      {screenRender()}
      <StatusBar style="auto" />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
