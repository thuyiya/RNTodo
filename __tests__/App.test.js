import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("<App />", () => {
  it("Contain With View", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.type).toBe('View');
  });

  it("No children outside of Provider", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children).toBe(null);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
