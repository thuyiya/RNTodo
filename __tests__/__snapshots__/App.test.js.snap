// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
    }
  }
/>
`;
